docs/docs/configuration/gstreamer.md

```diff
@@ -0,0 +1,324 @@
+---
+id: gstreamer
+title: GStreamer configuration
+---
+
+In addition to the FFmpeg, Frigate does support GStreamer. GStreamer is a framework for creating streaming media applications. The main advantages are that the pluggable components can be mixed and matched into arbitrary pipelines.
+GStreamer has better support of hardware-accelerated video decoding on NVidia Jetson devices.
+
+**Note:** There are no advantages of GStreamer versus FFmpeg on non-Jetson devices.
+
+## Minimal GStreamer Configuration
+
+```yaml
+camera_name:
+  gstreamer:
+    inputs:
+      - path: rtsp://10.0.0.1:554/stream0
+        roles:
+          - detect
+  detect:
+    width: 1920
+    height: 1080
+```
+
+With the minimal configuration GStreamer integration will do the following:
+- Run `gst-inspect-1.0` to get the available plugins
+- Run `gst-discoverer-1.0` against the RTSP stream. The discovery process gets the audio and video stream codecs
+- Build the GStreamer pipeline based on the available plugins.
+- GStreamer automatically enable audio stream for recording if audio is available
+
+The default configuration should be enough for most of the cases. However, if you have multiple cameras, running `gst-discoverer-1.0` for each one might be pretty time-consuming. To avoid running `gst-discoverer-1.0`, you have to specify the video codec as well as the audio codec for the `record` role.
+
+
+## GStreamer configuration with video and audio codecs set
+
+```yaml
+camera_name:
+  gstreamer:
+    video_format: video/x-h265
+    audio_format: audio/x-alaw
+    inputs:
+      - path: rtsp://10.0.0.1:554/stream0
+        # video_format: video/x-h265
+        # audio_format: audio/x-alaw
```

Comment:
Attempting to put `video_format` under `inputs` also yields an invalid config error.

Reply:
@yury-sannikov I see this is now commented out. Just re-confirming this continues to yield an invalid config error. This shorthand would be really nice. 

It'd also be nice if there was a shorthand added to the docs for ensuring `nvv4l2decoder` and `nvvidconv` are used.